app/src/main/java/org/wikipedia/page/edithistory/EditHistoryListViewModel.kt

```diff
@@ -123,8 +139,37 @@ class EditHistoryListViewModel(bundle: Bundle) : ViewModel() {
         override suspend fun load(params: LoadParams<String>): LoadResult<String, MwQueryPage.Revision> {
             return try {
                 val response = ServiceFactory.get(WikiSite.forLanguageCode(pageTitle.wikiSite.languageCode))
-                        .getRevisionDetailsDescending(pageTitle.prefixedText, params.loadSize, null, params.key)
-                LoadResult.Page(response.query!!.pages?.get(0)?.revisions!!, null, response.continuation?.rvContinuation)
+                    .getRevisionDetailsDescending(pageTitle.prefixedText, params.loadSize, null, params.key)
+
+                val revision = response.query!!.pages?.first()?.revisions!!
+
+                // The `ususers` parameter has a size limit of 50 usernames.
+                val userNamesList = revision.asSequence().filterNot { it.isAnon }.map { it.user }.distinct().chunked(50).toList()
+
+                // Combining two async API responses, finding out all bot usernames, and joining them into one list.
+                val botUserList = withContext(Dispatchers.IO) {
+                    val result = userNamesList.map {
+                        async { ServiceFactory.get(pageTitle.wikiSite).getUserInfoList(it.joinToString("|")) }
```

Comment:
This is a very costly and unnecessary API call. I believe we discussed that we will not be supporting filtering by Bot edits? This is not only because the API does not support it, but also because no one actually needs it.